test/search/test_organized_index.cpp

```diff
@@ -132,16 +135,10 @@ TEST (PCL, Organized_Neighbor_Search_Pointcloud_Nearest_K_Neighbour_Search)
 
     std::priority_queue<prioPointQueueEntry> pointCandidates;
 
-
-    // push all points and their distance to the search point into a priority queue - bruteforce approach.
-    for (std::size_t i = 0; i < cloudIn->points.size (); i++)
+    for (auto it = cloudIn->begin(); it != cloudIn->end(); ++it)
     {
-      double pointDist = ((cloudIn->points[i].x - searchPoint.x) * (cloudIn->points[i].x - searchPoint.x) +
-                          (cloudIn->points[i].y - searchPoint.y) * (cloudIn->points[i].y - searchPoint.y) +
-                          (cloudIn->points[i].z - searchPoint.z) * (cloudIn->points[i].z - searchPoint.z));
-
-      prioPointQueueEntry pointEntry (cloudIn->points[i], pointDist, i);
-
+      const auto pointDist = geometry::distance(*it, searchPoint);
+      prioPointQueueEntry pointEntry (*it, pointDist, it - cloudIn->begin());
```

Comment:
It's better to use `std::distance` instead of subtraction (here as well as elsewhere)